### PR TITLE
Content(star): Magnetar sprite + hazard + haze

### PR DESCRIPTION
**Pull request by RisingLeaf**
**Content (Artwork / Missions / Jobs)**

## Summary
I think at this point it has become clear that I aim to introduce more unique stars to ES, this time I present: The Magnetar!

For those of you who dont know what a magnetar is, a magnetar is a neutron star that has such a strong magnetic field, that if the sun was a magnetar all planets in the solar system would burn from radiation.
Magnetars also have the annoying habit of sending out a bunch of gamma radiation from time to time that could wipe out all civilization on earth in a fraction of a second.

So pretty nasty things.

Note: Because someone already wants to use this in a project they will publish soon™️, and because I think there should not be more than two or three in the game given their amount in our galaxy, I would like to ask not to use right away.

In the PR also included is a haze that looks good with the sprite and two hazards that should be used with this star.

Screenshot 2023-01-19 at 12 27 43 PM
Screenshot 2023-01-19 at 12 27 39 PM

References:
image
image
image

## Artwork Checklist
Note: I will upload this stuff as soon as I know that the sprites are good, because the render for the 2x sprite of the magnetar is an overnight thing.
 - [ ] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [ ] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: {{insert PR link}}
 - [ ] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}

### CODE REVIEW

**Review by beccabunny — CHANGES_REQUESTED** [review]
The sprite is very nice! The only thing I'd prefer to see improved is the inner part of the star, with the blue and white stripes, which seems to have an excessive contrast and looks a bit underdetailed compared to the outer beams. I really like the rest of it though

**Comment by Hurleveur** [comment]
Great stuff alr
But shouldnt the title be content(star) instead of feat? Caus I see feat as a code feature but idk

**Comment by RisingLeaf** [comment]
> The sprite is very nice! The only thing I'd prefer to see improved is the inner part of the star, with the blue and white stripes, which seems to have an excessive contrast and looks a bit underdetailed compared to the outer beams. I really like the rest of it though

A magnetar is rotating 1 time per second. I dont think you could see a lot of detail, if you still think I should change it, I will do so

**Comment by warp-core** [comment]
> > The sprite is very nice! The only thing I'd prefer to see improved is the inner part of the star, with the blue and white stripes, which seems to have an excessive contrast and looks a bit underdetailed compared to the outer beams. I really like the rest of it though
> 
> A magnetar is rotating 1 time per second. I dont think you could see a lot of detail, if you still think I should change it, I will do so

Magnetars actually tend to rotate more slowly than standard neutron stars or pulsars.

Why not do relative damage?
Why gravitational attraction? These sorts of stars aren't particularly massive.

**Comment by RisingLeaf** [comment]
> > > The sprite is very nice! The only thing I'd prefer to see improved is the inner part of the star, with the blue and white stripes, which seems to have an excessive contrast and looks a bit underdetailed compared to the outer beams. I really like the rest of it though
> > 
> > 
> > A magnetar is rotating 1 time per second. I dont think you could see a lot of detail, if you still think I should change it, I will do so
> 
> Magnetars actually tend to rotate more slowly than standard neutron stars or pulsars.
> 
> Why not do relative damage? Why gravitational attraction? These sorts of stars aren't particularly massive.

Normal neutron stars rotate 10 times per second, so yes they are slower.

I dont know, should it be relative damage? From a "scientific" point, a magnetar is doing the same damage to every shield, why should tougher shields get more damage?

The gravitational attraction is from the magnetic field, because I think most ships have magnetic parts to them.

**Comment by warp-core** [comment]
> The gravitational attraction is from the magnetic field, because I think most ships have magnetic parts to them.

The magnetic field is not directed inwards like a gravitational field, though.
It's actually a notable feature of magnetic fields that (as far as we know) it is not possible to have magnetic field lines converge at some point like it is with gravitation or electric fields.

**Comment by RisingLeaf** [comment]
> 

That is right, but I dont think we have the means to create a gravitational force that goes in a circle in the game right now. So should I remove it?

File: data/hazards.txt [hunk 1]
```diff
@@ -100,3 +100,22 @@ hazard "Small Black Hole Gravity"
...
+
+
+hazard "Magnetar Radiation"
+	"constant strength"
+	"range" 3000
+	weapon
+		"discharge damage" 0.5
+		"damage dropoff" 120 3000
+		"hull damage" 1
+		"energy damage" 20
```
Comment (1010todd) [inline data/hazards.txt:112]: Not sure how much sense hull damage makes here. How about disruption damage?

**Comment by 1010todd** [comment]
I agree with Becca that the contrast on the stripes might be a bit much. Also for details just a few extra shorter blurry stripes with different colors (darker blues, lighter blues) might help?

File: data/hazards.txt [hunk 2]
```diff
@@ -109,3 +109,22 @@ hazard "Neutron Star Magnetic Field"
...
+
+hazard "Magnetar Magnetic Field"
+	"constant strength"
+	"range" 1500
+	weapon
+		"gravitational"
+		"hit force" -10
+		"damage dropoff" 20 5100
+		"dropoff modifier" 0
+
```
Comment (RisingLeaf) [inline data/hazards.txt:130]:
```suggestion

```

File: data/hazards.txt [hunk 3]
```diff
@@ -109,3 +109,21 @@ hazard "Neutron Star Magnetic Field"
...
+		"energy damage" 20
+
+hazard "Magnetar Magnetic Field"
+	"constant strength"
+	"range" 1500
+	weapon
+		"gravitational"
+		"hit force" -10
+		"damage dropoff" 20 5100
+		"dropoff modifier" 0
```
Comment (RisingLeaf) [inline data/hazards.txt:129]:
```suggestion
		"dropoff modifier" 0

```

File: data/hazards.txt (hunk 1 above)
Comment (RisingLeaf) [reply data/hazards.txt:120]: I think that could work too

**Comment by RisingLeaf** [comment]
new magnetar version:
magnetar+

**Comment by Saugia** [comment]
It's looking better, but I can't help but think the polar jets are too straight-lined right now, and the ends of them are too flat. It looks less natural this way. But I'll grab @beccabunny and @1010todd again for their opinions too.

**Comment by 1010todd** [comment]
The jet on existing one is also quite straight but maybe the strong magnetic field could do something to it I guess. Yeah the end could be less flat and be slightly more jagged(Like, spike or two) or have rounded end like existing neutron star.

**Comment by Saugia** [comment]
Agree with everything Todd said, especially having more variance in surface patch sizing.

**Comment by RisingLeaf** [comment]
Just fyi, I am working on it, but for some reason(I am stupid that is the reason) the whole thing takes over an hour to render, so it is only going slow

**Comment by 1010todd** [comment]
> Just fyi, I am working on it, but for some reason(I am stupid that is the reason) the whole thing takes over an hour to render, so it is only going slow

What's the samples count for render? (in render properties). Try lowering that too 128 or something + denoising should be enough.

**Comment by RisingLeaf** [comment]
> > Just fyi, I am working on it, but for some reason(I am stupid that is the reason) the whole thing takes over an hour to render, so it is only going slow
> 
> What's the samples count for render? (in render properties). Try lowering that too 128 or something + denoising should be enough.

well that could be the issue(2048 default)

**Comment by Zitchas** [comment]
Looks beautiful! Would love to have something like this in the Ember Waste. Or, if you have any ideas for can't-possibly-exist-but-look-cool stars, the EW would be the place to have them. (especially in later chapters beyond Postverta...)

**Comment by RisingLeaf** [comment]
> Looks beautiful! Would love to have something like this in the Ember Waste. Or, if you have any ideas for can't-possibly-exist-but-look-cool stars, the EW would be the place to have them. (especially in later chapters beyond Postverta...)

While I am not against it there is already someone who wants this in a region that fits this Star really good so I would like to ask you to wait a bit

**Comment by RisingLeaf** [comment]
Also magnetars are not that seldom in our galaxy

**Comment by RisingLeaf** [comment]
here comes another try:
Screenshot 2023-02-06 at 5 10 27 PM
Screenshot 2023-02-06 at 5 10 43 PM

**Comment by RisingLeaf** [comment]
guess I will ping some people just because I am annoying:
@Saugia @1010todd

**Comment by 1010todd** [comment]
Quick look: The star surface looks better now, might be a slight bit too blurry and a bit too bright though, have to compare with regular neutron star.
The pulsar beam cutoff is still too even imo. I meant more like having a spike of brighter area a fair bit beyond adjacent dimmer area. The beam texture overall is also clumped into more visible blobs that looks more like aurora than a .5c beam.

Not sure how well I explained it so I might make some image to explain it later (6+ hours or something, likely more.).

**Comment by RisingLeaf** [comment]
Screenshot 2023-02-07 at 7 21 20 PM
Screenshot 2023-02-07 at 7 21 33 PM
Screenshot 2023-02-07 at 7 21 48 PM

spikier, I guess
am I getting closer?

**Comment by 1010todd** [comment]
Yeah, that's closer. The left side might be a bit too even though (circled in green). Also the beam coming out of the star(circled in red) might be a bit too broad now, the narrowness of the previous iteration looks fine.

image

The cutoff into empty space also seems to be a bit too sharp.

Also I just looked at the regular neutron star again and it does seems like the star of this one is too bright.

**Comment by RisingLeaf** [comment]
Another try:
Screenshot 2023-02-12 at 3 20 31 PM
Screenshot 2023-02-12 at 3 20 38 PM

**Comment by RisingLeaf** [comment]
tried to apply saugias feedback on stream:
Screenshot 2023-02-12 at 5 51 01 PM
Screenshot 2023-02-12 at 5 51 07 PM
@Saugia annyoing ping

**Comment by Saugia** [comment]
I'm still not sure about the middle cloud. The blue gradient is good, but I'm not sure about the stylistic haze (the part that looks like the two jet streams). Otherwise, the rest looks good now.

**Review by beccabunny — APPROVED** [review]
Looks great to me, I'd apply what Saugia asked and then it's good for merging

**Comment by 1010todd** [comment]
I prefer the more subtle inner haze (blue ones within the loops, perpendicular to jet stream) of the older one because I think the main focus should be the jets and the magnetic loop around the star, so maybe it should be a slight bit dimmer imo. Also the texture of the jet stream looks more like aurora again, some blur on the hardedges might help make it look more like the existing neutron star.
(Also maybe request review from Gefu because he made the current neutron star.)

**Pull request closed without merging**